charts/localstack/templates/deployment.yaml

```
@@ -70,7 +70,11 @@ spec:
             {{- end }}
             {{- if .Values.enableStartupScripts }}
               - name: {{ template "localstack.fullname" . }}-init-scripts-config
+              {{- if or (hasPrefix "0." .Values.image.tag) (hasPrefix "1." .Values.image.tag) }}
                 mountPath: /docker-entrypoint-initaws.d
+              {{- else }}
+                mountPath: /etc/localstack/init/ready.d
```

```suggestion
                mountPath: /etc/localstack/init/ready.d/{{ template "localstack.fullname" . }}-init-scripts-config.sh
                subPath: {{ template "localstack.fullname" . }}-init-scripts-config.sh
```

I have found that mounting the single file via subPath is less flaky.

Without subPath, the contents of the configMap are mounted as symlinks to ..data, and this can result in file-not-found the first time it is deployed.

The downside is that subPaths are not automatically updated when a ConfigMap is modified.

---

I am sorry, but this is not in the scope of this PR.
If you have seen issues with mounting the init script, please create a separate issue.